src/Build.UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -4741,6 +4741,59 @@ public void PropertyFunctionStringArrayGetValue()
             TestPropertyFunction("$(X.Split($([System.Convert]::ToString(`.`).ToCharArray())).GetValue($([System.Convert]::ToInt32(0))))", "X", "ab.cd", "ab");
         }
 
+        /// <summary>
+        /// Test that Char.IsDigit fast-path works correctly
+        /// </summary>
+        [Fact]
+        public void PropertyFunctionCharIsDigit()
```

Comment:
Rewrite this test to be a data-driven Theory and use the helper TestPropertyFunction.